src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/Internal/Patterns/PatternBuilder.cs

```diff
@@ -199,7 +199,7 @@ public IPattern Build(string pattern)
                 scanPattern = endSegment + 1;
             }
 
-            if (segmentsPatternStartsWith == null)
+            if (segmentsPatternStartsWith == null || segmentsPatternEndsWith == null || segmentsPatternContains == null)
```

Comment:
I don't think we should be changing behavior here. We will probably need to annotate `RaggedPattern` to allow for null `segmentsPatternEndsWith` and `segmentsPatternContains`, if they can't be proven to not be null here.